scylla-cdc-driver3/src/main/java/com/scylladb/cdc/cql/driver3/Driver3SchemaBuilder.java

```diff
@@ -178,40 +531,31 @@ private void generatePrimaryKeyColumns() {
                 return new ChangeSchema.DataType(ChangeSchema.CqlType.DURATION);
             case LIST:
                 Preconditions.checkArgument(typeArguments.size() == 1);
-                return new ChangeSchema.DataType(ChangeSchema.CqlType.LIST, typeArguments, driverType.isFrozen());
+                return new ChangeSchema.DataType(ChangeSchema.CqlType.LIST, typeArguments, true);
             case MAP:
                 Preconditions.checkArgument(typeArguments.size() == 2);
-                return new ChangeSchema.DataType(ChangeSchema.CqlType.MAP, typeArguments, driverType.isFrozen());
+                return new ChangeSchema.DataType(ChangeSchema.CqlType.MAP, typeArguments, true);
             case SET:
                 Preconditions.checkArgument(typeArguments.size() == 1);
-                return new ChangeSchema.DataType(ChangeSchema.CqlType.SET, typeArguments, driverType.isFrozen());
+                return new ChangeSchema.DataType(ChangeSchema.CqlType.SET, typeArguments, true);
             case UDT:
                 Preconditions.checkArgument(driverType instanceof UserType);
                 UserType userType = (UserType) driverType;
                 ImmutableMap.Builder<String, ChangeSchema.DataType> fieldsBuilder = new ImmutableMap.Builder<>();
                 for (UserType.Field f : userType) {
-                    fieldsBuilder.put(f.getName(), translateColumnDataType(f.getType()));
+                    fieldsBuilder.put(f.getName(), translateCdcColumnDataType(f.getType()));
                 }
                 ChangeSchema.DataType.UdtType udtType = new ChangeSchema.DataType.UdtType(fieldsBuilder.build(), userType.getKeyspace(), userType.getTypeName());
-                return new ChangeSchema.DataType(ChangeSchema.CqlType.UDT, udtType, driverType.isFrozen());
+                return new ChangeSchema.DataType(ChangeSchema.CqlType.UDT, udtType, true);
             case TUPLE:
                 Preconditions.checkArgument(driverType instanceof TupleType);
                 TupleType tupleType = (TupleType) driverType;
-                typeArguments = tupleType.getComponentTypes().stream().map(this::translateColumnDataType)
+                typeArguments = tupleType.getComponentTypes().stream().map(this::translateCdcColumnDataType)
                         .collect(Collectors.collectingAndThen(Collectors.toList(), ImmutableList::copyOf));
                 Preconditions.checkArgument(!typeArguments.isEmpty());
-                return new ChangeSchema.DataType(ChangeSchema.CqlType.TUPLE, typeArguments, driverType.isFrozen());
+                return new ChangeSchema.DataType(ChangeSchema.CqlType.TUPLE, typeArguments, true);
```

Comment:
A localised comment explaining that all CDC columns are frozen would be nice here.